src/NServiceBus.AzureStoragePersistence/Subscriptions/Azure/AzureSubscriptionStorage.cs

```diff
@@ -1,100 +1,103 @@
+            var table = client.GetTableReference(subscriptionTableName);
+
+            var encodedAddress = EncodeTo64(address.ToString());
+            foreach (var messageType in messageTypes)
             {
-                var encodedAddress = EncodeTo64(address.ToString());
-                foreach (var messageType in messageTypes)
+                var type = messageType;
+                var query = from s in table.CreateQuery<Subscription>()
+                    where s.PartitionKey == type.ToString() && s.RowKey == encodedAddress
+                    select s;
+                var subscription = query.AsTableQuery().AsEnumerable().SafeFirstOrDefault();
+                if (subscription != null)
                 {
-                    var type = messageType;
-                    var query = from s in context.Subscriptions
-                                where s.PartitionKey == type.ToString() && s.RowKey == encodedAddress
-                                select s;
-
-                    var subscription = query
-                        .AsTableServiceQuery(context) // Fixes #191
-                        .AsEnumerable() // Fixes #191, continuation not applied on single resultsets eventhough continuation can happen
-                        .SafeFirstOrDefault();
-                    if(subscription != null) context.DeleteObject(subscription);
-                    context.SaveChangesWithRetries();
+                    var operation = TableOperation.Delete(subscription);
+                    table.Execute(operation);
                 }
             }
         }
 
 
-
         IEnumerable<Address> ISubscriptionStorage.GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
         {
             var subscribers = new List<Address>();
+            var table = client.GetTableReference(subscriptionTableName);
 
-            using (var context = new SubscriptionServiceContext(client))
+            foreach (var messageType in messageTypes)
             {
-                foreach (var messageType in messageTypes)
-                {
-                    var type = messageType;
-                    var query = from s in context.Subscriptions
-                                where s.PartitionKey == type.ToString() 
-                                select s;
+                var type = messageType;
+                var query = from s in table.CreateQuery<Subscription>()
+                    where s.PartitionKey == type.ToString()
+                    select s;
 
-                    var result = query
-                        .AsTableServiceQuery(context) // Fixes #191
-                        .ToList();
+                var result = query.ToList();
 
-                    subscribers.AddRange(result.Select(s => Address.Parse(DecodeFrom64(s.RowKey))));
-                }
+                subscribers.AddRange(result.Select(s => Address.Parse(DecodeFrom64(s.RowKey))));
```

Comment:
i dont think the above tolist is required if u r just enumerating them. may as well save some memory and do

    
                subscribers.AddRange(query.Select(s => Address.Parse(DecodeFrom64(s.RowKey))));